src/operator/tensor/elemwise_unary_op.h

```diff
@@ -578,12 +458,50 @@ void CastStorageComputeEx(const nnvm::NodeAttrs& attrs,
   .set_num_outputs(1)                                               \
   .set_attr<nnvm::FInferShape>("FInferShape", ElemwiseShape<1, 1>)  \
   .set_attr<nnvm::FInferType>("FInferType", ElemwiseType<1, 1>)     \
+  .set_attr<nnvm::FInferStorageType>("FInferStorageType", ElemwiseStorageType<1, 1>) \
   .set_attr<nnvm::FInplaceOption>("FInplaceOption",                 \
     [](const NodeAttrs& attrs){                                     \
       return std::vector<std::pair<int, int> >{{0, 0}};             \
     })                                                              \
   .add_argument("data", "NDArray-or-Symbol", "The input array.")
 
+#define MXNET_OPERATOR_REGISTER_UNARY_DR(name)                      \
+  NNVM_REGISTER_OP(name)                                            \
+  .set_num_inputs(1)                                                \
+  .set_num_outputs(1)                                               \
+  .set_attr<nnvm::FInferShape>("FInferShape", ElemwiseShape<1, 1>)  \
+  .set_attr<nnvm::FInferType>("FInferType", ElemwiseType<1, 1>)     \
+  .set_attr<nnvm::FInplaceOption>("FInplaceOption",                 \
+    [](const NodeAttrs& attrs){                                     \
+      return std::vector<std::pair<int, int> >{{0, 0}};             \
+    })                                                              \
+  .add_argument("data", "NDArray-or-Symbol", "The input array.")
+
+/*! \brief Unary launch */
+#define MXNET_OPERATOR_REGISTER_UNARY_LAUNCH(name, __xpu$, __kernel$)                   \
+  MXNET_OPERATOR_REGISTER_UNARY(name)                                                   \
+  .set_attr<FCompute>("FCompute<" #__xpu$ ">", UnaryOp::Launch<__xpu$, __kernel$>)      \
+  .set_attr<FComputeEx>("FComputeEx<" #__xpu$ ">", UnaryOp::LaunchEx<__xpu$, __kernel$>)
+
+/*! \brief Unary launch, dense result */
+#define MXNET_OPERATOR_REGISTER_UNARY_LAUNCH_DR(name, __xpu$, __kernel$)                \
+  MXNET_OPERATOR_REGISTER_UNARY_DR(name)                                                \
+  .set_attr<FCompute>("FCompute<" #__xpu$ ">", UnaryOp::Launch<__xpu$, __kernel$>)      \
+  .set_attr<FComputeEx>("FComputeEx<" #__xpu$ ">", UnaryOp::LaunchAsDense<__xpu$, __kernel$>)
```

Comment:
If you LaunchAsDense it always creates a dense copy of the input. We can also write separate kernels that handles sparse input and generate dense output, which avoids this copy. But since writing separate kernels is time consuming, it's fine for now just to convert input to dense. We probably want to optimize it later if it's not performant.